wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -569,10 +562,10 @@ abstract class Wallet
       }
     }
     val newAccountDb = AccountDb(xpub, hdAccount)
-    val accountCreationF = accountDAO.create(newAccountDb)
-    accountCreationF.map(created =>
-      logger.debug(s"Created new account ${created.hdAccount}"))
-    accountCreationF.map(_ => this)
+    accountDAO.create(newAccountDb).map { created =>
```

Comment:
Things like this don't really help in my opinion, I don't think our solution should be "let's NOT use intermediate `val`s to avoid this problem!" This makes code unreadable.

Reply:
We don't need intermediate vals in most of the cases, because they create the unnecessary parallelization